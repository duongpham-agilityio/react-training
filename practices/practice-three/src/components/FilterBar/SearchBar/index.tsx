import { ChangeEvent, memo, useCallback } from 'react';
import { Center, Input, InputGroup } from '@chakra-ui/react';

// Icons
import { Search } from '@/assets/icons';

export type TSearchBarProps = {
  value: string;
  onChange: (value: string) => void;
};

export const SearchBar = memo(({ value, onChange }: TSearchBarProps) => {
  const handleChangeValue = useCallback(
    (e: ChangeEvent<HTMLInputElement>) => {
      const value = e.target.value;

      onChange(value);
    },
    [onChange],
  );

  return (
    <InputGroup
      border="1px"
      borderColor="gray.300"
      alignItems="center"
      borderRadius="md"
    >
      <Input
        fontSize="lg"
        placeholder="Search"
        borderColor="transparent"
        focusBorderColor="transparent"
        _hover={{
          borderColor: 'transparent',
        }}
        _placeholder={{
          fontWeight: 'light',
        }}
        value={value}
        onChange={handleChangeValue}
      />
      <Center h="60px" pr={5}>
        <Search />
      </Center>
    </InputGroup>
  );
});
