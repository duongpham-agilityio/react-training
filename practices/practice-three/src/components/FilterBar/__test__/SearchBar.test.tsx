import { fireEvent, render } from '@testing-library/react';

// Components
import { SearchBar, SearchBarProps } from '../SearchBar';

const onChange = jest.fn();
const defaultProps: SearchBarProps = {
  value: '',
  onChange,
};
const setup = (props = defaultProps) => render(<SearchBar {...props} />);

describe('SearchBar', () => {
  it('Match to snapshot', () => {
    const { container } = setup();

    expect(container).toMatchSnapshot();
  });

  it('Change value', () => {
    const { getByPlaceholderText } = setup();
    const input: HTMLInputElement = getByPlaceholderText(
      'Search',
    ) as HTMLInputElement;

    fireEvent.input(input, { target: { value: 'a' } });

    expect(onChange).toBeCalled();
  });
});
