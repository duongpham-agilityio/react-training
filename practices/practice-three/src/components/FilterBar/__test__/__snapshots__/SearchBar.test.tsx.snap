// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar Match to snapshot 1`] = `
<div>
  <div
    class="chakra-input__group css-viwvyh"
    data-group="true"
  >
    <input
      class="chakra-input css-190aajt"
      placeholder="Search"
      value=""
    />
    <div
      class="css-1q3ec5g"
    >
      <svg
        fill="none"
        height="30"
        width="42"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M25.507 7.2H1.8a1.8 1.8 0 0 1 0-3.6h23.707a5.402 5.402 0 0 1 10.186 0H40.2a1.8 1.8 0 0 1 0 3.6h-4.507a5.402 5.402 0 0 1-10.186 0ZM40.2 26.4a1.8 1.8 0 0 0 0-3.6H16.493a5.402 5.402 0 0 0-10.186 0H1.8a1.8 1.8 0 0 0 0 3.6h4.507a5.402 5.402 0 0 0 10.186 0H40.2Z"
          fill="#6A6A6A"
        />
      </svg>
    </div>
  </div>
</div>
`;
